README.md
# GLPI, plugin Monitoring

[![Build Status](https://travis-ci.org/ddurieux/glpi_monitoring.svg?branch=master)](https://travis-ci.org/ddurieux/glpi_monitoring)
[![Coverage Status](https://coveralls.io/repos/ddurieux/glpi_monitoring/badge.svg)](https://coveralls.io/r/ddurieux/glpi_monitoring)

A Glpi plugin to monitor computers, network equipments and printers with Shinken

##Don't know Glpi?

... then you may learn about it! [Check out Glpi](http://www.glpi-project.org/)!

##Don't know Shinken?

... then you may learn about it! [Check out Shinken](http://www.shinken-monitoring.org/)!

##Features

Monitoring plugin for Glpi allows to monitor managed computers:

* define Shinken configuration : 
 - checks, commands, contacts, calendars
 - components as the combination of commands, checks, contacts and graph templates for performance data
 - components catalogs as the association between components and hosts
 - business rules (known as services catalogs)
* collect and consult checks results through a dashboard : 
 - hosts states
 - resources (Shinken services) states
 - services catalogs (Shinken business rules)
 - graph performance data
 - view performance data counters
* acknowledge problems and associates with Glpi tickets managed in helpdesk


## Info
	Please see the [plugin page](https://forge.indepnet.net/projects/monitoring/)
  
## License
	------------------------------------------------------------------------
	Plugin Monitoring for GLPI
	Copyright (C) 2011-2015 by the Plugin Monitoring for GLPI Development Team.

	https://forge.indepnet.net/projects/monitoring/
	------------------------------------------------------------------------

	Plugin Monitoring for GLPI is free software: you can redistribute it and/or modify
	it under the terms of the GNU Affero General Public License as published by
	the Free Software Foundation, either version 3 of the License, or
	(at your option) any later version.

	Plugin Monitoring for GLPI is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
	GNU Affero General Public License for more details.

	You should have received a copy of the GNU Affero General Public License
	along with Monitoring. If not, see <http://www.gnu.org/licenses/>.
	------------------------------------------------------------------------


CHANGES
Changelog
=========

Here you can see the full list of changes between each release.

In Development
--------------

- Fix: PHP notice on Shinken generator

Version 0.85+1.3
~~~~~~~~~~~~~~~~

- Fix: Return of the Travis for unit tests.
- Fix: Errors on install / update plugin (PHP errors, SQL errors)
- Fix: Compatibility of plugin for PHP 5.3
- Fix: PHP errors in services catalogs
- Fix: Business priority in components and services catalogs from 0 to 5 
  instead 1 to 5
- Fix: Errors in Shinken configuration generator
- Fix: Problem with component argument contain '[[IP]]' string (not converted)
- Enhanced: Better performances to generate shinken configuration
- Enhanced: Add many tests for Shinken configuration generator
- Enhanced: Install script in CLI
- Enhanced: Add option in user form to redirect home (when log in GLPI) to 
  monitoring dashboard (end-user interface only)
- Enhanced: Disable move action of items in dashboard views when not in 
  edit mode
- Enhanced: Add columns in components catalogs and possibility to modify them
  in massive actions


Version 0.85+1.2
~~~~~~~~~~~~~~~~

- Fix: Save button in preferences for graph form
- Fix: Invert values in graphs
- Fix: Graphs with min and max values
- Fix: Timezone of graphs for browsers under Windows
- Fix: Bug in limit of components catalogs rules (not get all elements to put 
  in monitoring)
- Fix: Minemap for components catalogs
- Fix: Right to have link in counters of components catalogs
- Fix: Fil d'ariane in rules page (components catalogs and views)
- Fix: Massive actions in target tab of view and sliders
- Fix: Rules of view && delete button
- Fix: Delete item in view
- Fix: Copy components and commands.
- Fix: Javascript in services catalogs
- Fix: Problem with business rules (services catalogs)
- Fix: PHP errors when GLPI try to communicate with Shinken on port 7760
- Fix: Errors in some cases when generate Shinken configuration
- Fix: Right for restart Shinken
- Fix: Javascript problem in Weathermap configuration
- Fix: Link of node in Weathermap
- Fix: Fatal error when update plugin
- Fix: Use host entity calendar for host check_period
- Fix: Rights and display monitoring in simplified interface
- Fix: Date format in lists of dashboards (hosts status, resources status)
- Fix: Cron added 2 times in some cases (when update)
- Fix: Fatal error on check networkport of switch for monitoring
- Enhanced: Add counter of contacts in tab in components catalogs
- Enhanced: Add notifications options for host and components catalogs
- Enhanced: Recoded Shinken configuration generator
- Enhanced: Add Shinken state table
- Enhanced: Add script to create a GLPI ticket from Shinken
- Enhanced: Add code to write all key=>value of data to send to Shinken in 
  the right type (so better configuration generation)
- Enhanced: Display preferences for graph
- Enhanced: Reduce number of MySQL queries when display monitoring messages
- Enhanced: Add graphite prefix
- Enhanced: Add business priority for components
- Enhanced: Performances to list hosts status in dashboard
- Enhanced: All links in view opened now in new tab
- Enhanced: Get all timeperiods in end when generate configuration


Version 0.85+1.1
~~~~~~~~~~~


setup.php
<?php

/*
   ------------------------------------------------------------------------
   Plugin Monitoring for GLPI
   Copyright (C) 2011-2015 by the Plugin Monitoring for GLPI Development Team.

   https://forge.indepnet.net/projects/monitoring/
   ------------------------------------------------------------------------

   LICENSE

   This file is part of Plugin Monitoring project.

   Plugin Monitoring for GLPI is free software: you can redistribute it and/or modify
   it under the terms of the GNU Affero General Public License as published by
   the Free Software Foundation, either version 3 of the License, or
   (at your option) any later version.

   Plugin Monitoring for GLPI is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
   GNU Affero General Public License for more details.

   You should have received a copy of the GNU Affero General Public License
   along with Monitoring. If not, see <http://www.gnu.org/licenses/>.

   ------------------------------------------------------------------------

   @package   Plugin Monitoring for GLPI
   @author    David Durieux
   @co-author
   @comment
   @copyright Copyright (c) 2011-2015 Plugin Monitoring for GLPI team
   @license   AGPL License 3.0 or (at your option) any later version
              http://www.gnu.org/licenses/agpl-3.0-standalone.html
   @link      https://forge.indepnet.net/projects/monitoring/
   @since     2011

   ------------------------------------------------------------------------
 */

define ("PLUGIN_MONITORING_VERSION","0.85+1.3");

define('_MPDF_TEMP_PATH', GLPI_PLUGIN_DOC_DIR.'/monitoring/pdf/');

// Used for use config values in 'cache'
$PM_CONFIG = array();


// Init the hooks of monitoring
function plugin_init_monitoring() {
   global $PLUGIN_HOOKS;

   $PLUGIN_HOOKS['csrf_compliant']['monitoring'] = true;

//   $PLUGIN_HOOKS['change_profile']['monitoring'] = array('PluginMonitoringProfile','changeprofile');

   $Plugin = new Plugin();
   if ($Plugin->isActivated('monitoring')) {

         Plugin::registerClass('PluginMonitoringEntity',
              array('addtabon' => array('Entity')));
         Plugin::registerClass('PluginMonitoringCommmand');
         Plugin::registerClass('PluginMonitoringEventhandler');
         Plugin::registerClass('PluginMonitoringComponent');
         //Plugin::registerClass('PluginMonitoringComponentscatalog');
         Plugin::registerClass('PluginMonitoringComponentscatalog',
              array('addtabon' => array('Central')));
         Plugin::registerClass('PluginMonitoringContact',
              array('addtabon' => array('User')));
         Plugin::registerClass('PluginMonitoringDisplayview',
              array('addtabon' => array('Central')));
         Plugin::registerClass('PluginMonitoringHost',
              array('addtabon' => array('Central', 'Computer', 'Device', 'Printer', 'NetworkEquipment')));
         Plugin::registerClass('PluginMonitoringService',
              array('addtabon' => array('Central')));
         Plugin::registerClass('PluginMonitoringProfile',
              array('addtabon' => array('Profile')));
         Plugin::registerClass('PluginMonitoringServicescatalog',
              array('addtabon' => array('Central')));
         Plugin::registerClass('PluginMonitoringUnavailability',
              array('addtabon' => array('Computer', 'NetworkEquipment')));
         Plugin::registerClass('PluginMonitoringSystem',
              array('addtabon' => array('Central')));
         Plugin::registerClass('PluginMonitoringDowntime',
              array('addtabon' => array('Computer', 'Ticket')));
         Plugin::registerClass('PluginMonitoringAcknowledge',
              array('addtabon' => array('Computer')));
         Plugin::registerClass('PluginMonitoringHostdailycounter',
              array('addtabon' => array('Computer')));
         Plugin::registerClass('PluginMonitoringServiceevent',
              array('addtabon' => array('Computer')));
         Plugin::registerClass('PluginMonitoringRedirecthome',
              array('addtabon' => array('User')));
         // Plugin::registerClass('PluginMonitoringHostCounter');

         if (class_exists('PluginAppliancesAppliance')) {
           PluginAppliancesAppliance::registerType('PluginMonitoringServicescatalog');
         }


         $PLUGIN_HOOKS['use_massive_action']['monitoring']=1;
         $PLUGIN_HOOKS['add_css']['monitoring'] = array(
            "lib/nvd3/src/nv.d3.css",
            "lib/jqueryplugins/tagbox/css/jquery.tagbox.css",
            "css/views.css"
            );
         $PLUGIN_HOOKS['add_javascript']['monitoring'] = array(
             "lib/jscolor/jscolor.js",
             "lib/jqueryplugins/tagbox/js/jquery.tagbox.js"
             );

         $plugin = new Plugin();
         if ($plugin->isActivated('monitoring')) {
            if (isset($_SESSION["glpiactiveprofile"])
                    && isset($_SESSION["glpiactiveprofile"]["interface"])
                    && $_SESSION["glpiactiveprofile"]["interface"] == "helpdesk") {
               $profile = new Profile();
               if ($profile->getFromDB($_SESSION['glpiactiveprofile']['id'])) {
                  $prof = array();
                  foreach ($profile->fields as $rname=>$right) {
                     if (substr($rname, 0, 18) === 'plugin_monitoring_') {
                        $_SESSION['glpiactiveprofile'][$rname] = $right;
                     }
                  }
               }
            }

            $PLUGIN_HOOKS['menu_toadd']['monitoring'] = array('plugins' => 'PluginMonitoringDashboard');
            if (Session::haveRight("plugin_monitoring_dashboard", READ)) {
               $PLUGIN_HOOKS["helpdesk_menu_entry"]['monitoring'] = '/front/dashboard.php';
            }
         }

         $PLUGIN_HOOKS['config_page']['monitoring'] = 'front/config.form.php';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['config'] = 'front/config.form.php';

         // Tabs for each type
         $PLUGIN_HOOKS['headings']['monitoring'] = 'plugin_get_headings_monitoring';
         $PLUGIN_HOOKS['headings_action']['monitoring'] = 'plugin_headings_actions_monitoring';

         // Icons add, search...
         // Still useful to declare all that stuff ? Menu is ok without this ...
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['command'] = 'front/command.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['command'] = 'front/command.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['checks'] = 'front/check.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['checks'] = 'front/check.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['componentscatalog'] = 'front/componentscatalog.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['componentscatalog'] = 'front/componentscatalog.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['servicescatalog'] = 'front/servicescatalog.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['servicescatalog'] = 'front/servicescatalog.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['components'] = 'front/component.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['components'] = 'front/component.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['contacttemplates'] = 'front/contacttemplate.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['contacttemplates'] = 'front/contacttemplate.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['hostnotificationtemplates'] = 'front/hostnotificationtemplate.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['hostnotificationtemplates'] = 'front/hostnotificationtemplate.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['servicenotificationtemplates'] = 'front/servicenotificationtemplate.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['servicenotificationtemplates'] = 'front/servicenotificationtemplate.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['displayview'] = 'front/displayview.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['displayview'] = 'front/displayview.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['PluginMonitoringRealm'] = 'front/realm.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['PluginMonitoringRealm'] = 'front/realm.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['weathermap'] = 'front/weathermap.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['weathermap'] = 'front/weathermap.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['eventhandler'] = 'front/eventhandler.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['eventhandler'] = 'front/eventhandler.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['notificationcommand'] = 'front/notificationcommand.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['notificationcommand'] = 'front/notificationcommand.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['perfdata'] = 'front/perfdata.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['perfdata'] = 'front/perfdata.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['customitem_gauge'] = 'front/customitem_gauge.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['customitem_gauge'] = 'front/customitem_gauge.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['customitem_counter'] = 'front/customitem_counter.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['customitem_counter'] = 'front/customitem_counter.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['add']['slider'] = 'front/slider.form.php?add=1';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['slider'] = 'front/slider.php';

         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['service'] = 'front/display.php';
         $PLUGIN_HOOKS['submenu_entry']['monitoring']['search']['service'] = 'front/host.php';

         if (isset($_SESSION["glpiname"])) {

            // Fil ariane
            // Still useful to declare all that stuff ? Menu is ok without this ...
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['check']['title'] = __('Check definition', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['check']['page']  = '/plugins/monitoring/front/check.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['command']['title'] = __('Commands', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['command']['page']  = '/plugins/monitoring/front/command.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['components']['title'] = __('Components', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['components']['page']  = '/plugins/monitoring/front/component.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['componentscatalog']['title'] = __('Components catalog', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['componentscatalog']['page']  = '/plugins/monitoring/front/componentscatalog.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['contacttemplates']['title'] = __('Contact templates', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['contacttemplates']['page']  = '/plugins/monitoring/front/contacttemplate.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['hostnotificationtemplates']['title'] = __('Hosts notifications templates', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['hostnotificationtemplates']['page']  = '/plugins/monitoring/front/hostnotificationtemplate.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['servicenotificationtemplates']['title'] = __('Services notifications templates', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['servicenotificationtemplates']['page']  = '/plugins/monitoring/front/servicenotificationtemplate.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['display']['title'] = __('Dashboard', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['display']['page']  = '/plugins/monitoring/front/display_servicescatalog.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['displayview']['title'] = __('Views', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['displayview']['page']  = '/plugins/monitoring/front/displayview.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['PluginMonitoringRealm']['title'] = __('Reamls', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['PluginMonitoringRealm']['page']  = '/plugins/monitoring/front/realm.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['servicescatalog']['title'] = __('Services catalog', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['servicescatalog']['page']  = '/plugins/monitoring/front/servicescatalog.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['weathermap']['title'] = __('Weathermap', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['weathermap']['page']  = '/plugins/monitoring/front/weathermap.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['eventhandler']['title'] = __('Event handler', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['eventhandler']['page']  = '/plugins/monitoring/front/eventhandler.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['notificationcommand']['title'] = __('Notification commands', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['notificationcommand']['page']  = '/plugins/monitoring/front/notificationcommand.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['hostdailycounter']['title'] = __('Host daily counters', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['hostdailycounter']['page']  = '/plugins/monitoring/front/hostdailycounter.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['unavailability']['title'] = __('Unavailabilities', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['unavailability']['page']  = '/plugins/monitoring/front/unavailability.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['downtime']['title'] = __('Downtimes', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['downtime']['page']  = '/plugins/monitoring/front/downtime.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['acknowledge']['title'] = __('Acknowledges', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['acknowledge']['page']  = '/plugins/monitoring/front/acknowledge.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['perfdata']['title'] = __('Graph templates', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['perfdata']['page']  = '/plugins/monitoring/front/perfdata.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['slider']['title'] = __('Carrousel / slider', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['slider']['page']  = '/plugins/monitoring/front/slider.php';

            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['dashboard']['title'] = __('Dashboard', 'monitoring');
            $PLUGIN_HOOKS['submenu_entry']['monitoring']['options']['dashboard']['page']  = '/plugins/monitoring/front/dashboard.php';
         }
         $rule_check = array('PluginMonitoringComponentscatalog_rule','isThisItemCheckRule');
         $rule_check_networkport = array('PluginMonitoringComponentscatalog_rule', 'isThisItemCheckRuleNetworkport');
         $PLUGIN_HOOKS['item_add']['monitoring'] =
                                 array('Computer'         => $rule_check,
                                       'NetworkEquipment' => $rule_check,
                                       'Printer'          => $rule_check,
                                       'Peripheral'       => $rule_check,
                                       'Phone'            => $rule_check,
                                       'PluginMonitoringNetworkport' => $rule_check_networkport,
                                       'PluginMonitoringComponentscatalog_rule' =>
                                             array('PluginMonitoringComponentscatalog_rule','getItemsDynamicly'),
                                       'PluginMonitoringComponentscatalog_Host' =>
                                             array('PluginMonitoringHost','addHost'));
         $PLUGIN_HOOKS['item_update']['monitoring'] =
                                 array('Computer'         => $rule_check,
                                       'NetworkEquipment' => $rule_check,
                                       'Printer'          => $rule_check,
                                       'Peripheral'       => $rule_check,
                                       'Phone'            => $rule_check,
                                       'PluginMonitoringComponentscatalog' =>
                                             array('PluginMonitoringComponentscatalog','replayRulesCatalog'),
                                       'PluginMonitoringComponentscatalog_rule' =>
                                             array('PluginMonitoringComponentscatalog_rule','getItemsDynamicly'));
         // No more hosts counters ...
         // $PLUGIN_HOOKS['pre_item_update']['monitoring'] =
                                 // array('PluginMonitoringHostdailycounter' =>
                                             // array('PluginMonitoringHostdailycounter','pre_item_update')
                                 // );

         $PLUGIN_HOOKS['item_purge']['monitoring'] =
                                 array('Computer'         => $rule_check,
                                       'NetworkEquipment' => $rule_check,
                                       'Printer'          => $rule_check,
                                       'Peripheral'       => $rule_check,
                                       'Phone'            => $rule_check,
                                       'NetworkPort'      => array('PluginMonitoringNetworkport', 'deleteNetworkPort'),
                                       'PluginMonitoringNetworkport' => $rule_check_networkport,
                                       'PluginMonitoringComponentscatalog_rule' =>
                                             array('PluginMonitoringComponentscatalog_rule','getItemsDynamicly'),
                                       'PluginMonitoringComponentscatalog_Host' =>
                                             array('PluginMonitoringComponentscatalog_Host','unlinkComponentsToItem'),
                                       'PluginMonitoringComponentscatalog' =>
                                             array('PluginMonitoringComponentscatalog','removeCatalog'),
                                       'PluginMonitoringBusinessrulegroup' =>
                                             array('PluginMonitoringBusinessrule','removeBusinessruleonDeletegroup'));

         if (!isset($_SESSION['glpi_plugin_monitoring']['_refresh'])) {
            $_SESSION['glpi_plugin_monitoring']['_refresh'] = '60';
         }
//      }
         $PLUGIN_HOOKS['post_init']['monitoring'] = 'plugin_monitoring_postinit';
      $PLUGIN_HOOKS['webservices']['monitoring'] = 'plugin_monitoring_registerMethods';

   }
   return $PLUGIN_HOOKS;
}

// Name and Version of the plugin
function plugin_version_monitoring() {
   return array('name'           => 'Monitoring',
                'shortname'      => 'monitoring',
                'version'        => PLUGIN_MONITORING_VERSION,
                'license'        => 'AGPLv3+',
                'author'         =>'<a href="mailto:d.durieux@siprossii.com">David DURIEUX</a>',
                'homepage'       =>'https://forge.indepnet.net/projects/monitoring/',
                'minGlpiVersion' => '0.85'
   );
}


// Optional : check prerequisites before install : may print errors or add to message after redirect
function plugin_monitoring_check_prerequisites() {

   if (version_compare(GLPI_VERSION,'0.85','lt') || version_compare(GLPI_VERSION,'0.86','ge')) {
      echo "error, require GLPI 0.85.x";
   } else {
      return true;
   }
}

function plugin_monitoring_check_config() {
   return true;
}

function plugin_monitoring_haveTypeRight($type,$right) {
   return true;
}

?>
